public/react-services/pattern-handler.js

```diff
@@ -22,80 +22,91 @@ export class PatternHandler {
    */
   static async getPatternList(where) {
     try {
-      var patternList = await SavedObject.getListOfWazuhValidIndexPatterns();
+      const wazuhConfig = new WazuhConfig();
+      const { pattern } = wazuhConfig.getConfig();
+
+      const defaultPatterns = [AppState.getCurrentPattern(), pattern];
+      let patternList = await SavedObject.getListOfWazuhValidIndexPatterns(defaultPatterns, where);
 
       if (where === 'healthcheck') {
         function getIndexPatterns() {
           return new Promise(function (resolve, reject) {
             setTimeout(async function () {
-              var patternList = await SavedObject.getListOfWazuhValidIndexPatterns();
+              const patternList = await SavedObject.getListOfWazuhValidIndexPatterns(
+                defaultPatterns,
+                where
+              );
               resolve(patternList);
             }, 500);
           });
         }
-        var i = 0;
+
+        let i = 0;
         // if the index pattern doesn't exist yet, we check 5 more times with a delay of 500ms
         while (i < 5 && !patternList.length) {
           i++;
-          patternList = await getIndexPatterns()
-            .then(
-              function (result) {
-                return result;
-              }
-            );
+          patternList = await getIndexPatterns().then(function (result) {
+            return result;
+          });
         }
-      }
 
-      const wazuhConfig = new WazuhConfig();
-      const { pattern } = wazuhConfig.getConfig();
-      const indexPatternFound = patternList.find((indexPattern) => indexPattern.title === pattern);
+        const indexPatternFound = patternList.find(
+          (indexPattern) => indexPattern.title === pattern
+        );
 
-      if (!indexPatternFound) {
-        // if no valid index patterns are found we try to create the wazuh-alerts-*
-        try {
+        if (!indexPatternFound) {
+          // if no valid index patterns are found we try to create the wazuh-alerts-*
+          try {
+            if (!pattern) return;
 
-          if (!pattern) return;
+            getToasts().add({
+              color: 'warning',
+              title: `No ${pattern} index pattern was found, proceeding to create it.`,
+              toastLifeTimeMs: 5000,
+            });
 
-          getToasts().add({
-            color: 'warning',
-            title:
-              `No ${pattern} index pattern was found, proceeding to create it.`,
-            toastLifeTimeMs: 5000
-          });
+            if (!indexPatternFound.length) {
```

Comment:
indexPatternFound is a object result of the patternList.find
length is not a valid property